### DISCUSSION TITLE
how to debug in changed crazyflie-firmware?

### DISCUSSION THREAD

**Discussion by powehi2020** [bitcraze/discussions#713] [Q&A]
I am a green hands.I try to change the controller in the true cf2.0, I change crazyflie-firmware code one time then make cload cf2.0 to test one time. It is too low efficiency. Can you give me some advice to enhance my efficiency? thk u ~

**Comment by knmcguire** [comment]
I think you are interested in onchip debugging? https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/openocd_gdb_debugging/. Or perhaps you can read this blogpost to learn more about other debug tools: https://www.bitcraze.io/2022/05/debug-tools-in-the-client-console-tab/

If this is not what you are looking for, you need to reformulate your question as it is quite unclear.
